src/docker-in-docker/install.sh

```diff
@@ -469,32 +469,24 @@ if [ "${INSTALL_DOCKER_BUILDX}" = "true" ]; then
     find "${docker_home}" -type d -print0 | xargs -n 1 -0 chmod g+s
 fi
 
-if [ -z "$DISABLE_IP6_TABLES" ]; then
-    DOCKER_DEFAULT_IP6_TABLES=""
-else
-    version_to_check=""
-    semver_regex="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?$"
-    if echo "$DOCKER_VERSION" | grep -Eq "$semver_regex"; then
-        version_to_check=$(echo $DOCKER_VERSION | cut -d. -f1)
-    elif echo "$DOCKER_VERSION" | grep -Eq "^-?[0-9]+$"; then
-        version_to_check=$DOCKER_VERSION
-    fi
+DOCKER_DEFAULT_IP6_TABLES=""
+version_to_check=""
+semver_regex="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?$"
+if echo "$DOCKER_VERSION" | grep -Eq "$semver_regex"; then
+    version_to_check=$(echo $DOCKER_VERSION | cut -d. -f1)
+elif echo "$DOCKER_VERSION" | grep -Eq "^-?[0-9]+$"; then
+    version_to_check=$DOCKER_VERSION
+fi
 
-    if [ -n "$version_to_check" ] && [ "$version_to_check" -ge 27 ] || [ "$DOCKER_VERSION" = "latest" ]; then
-            if [ "$DISABLE_IP6_TABLES" == false ]; then
-                DOCKER_DEFAULT_IP6_TABLES=""
-            else
-                echo "Info: Ip6tables feature is enabled by default in docker engine v27 & later. Disabling.."
-                DOCKER_DEFAULT_IP6_TABLES="--ip6tables=false"
-            fi
-    else 
+if [[ -n "$version_to_check" && "$version_to_check" -ge 27 ]] || [ "$DOCKER_VERSION" = "latest" ]; then
         if [ "$DISABLE_IP6_TABLES" == true ]; then
-            echo "Info: Ip6tables feature is already disabled in docker engine v26 & lower."
-            DOCKER_DEFAULT_IP6_TABLES=""
-        else
-            echo "Error: Ip6tables in docker engine v26 & lower is an experimental feature. Enabling is not supported."
-            exit 1
+            echo "Info: Ip6tables feature is enabled by default in docker engine v27 & later. Disabling.."
+            DOCKER_DEFAULT_IP6_TABLES="--ip6tables=false"
         fi
+else 
+    if [ "$DISABLE_IP6_TABLES" == false ]; then
+        echo "Error: Ip6tables in docker engine v26 & lower is an experimental feature. Enabling is not supported. Aborting.."
+        exit 1
```

Comment:
Is it possible to add tests for Docker >= 27 and `disableIp6tables:true` ?

Reply:
it's already there `dockerIp6tablesDisabledTest`